boards/BMS/Src/App/states/App_ChargeState.c

```diff
@@ -25,48 +29,77 @@ static void ChargeStateRunOnTick100Hz(struct StateMachine *const state_machine)
 {
     if (App_AllStatesRunOnTick100Hz(state_machine))
     {
-        struct BmsWorld *   world       = App_SharedStateMachine_GetWorld(state_machine);
-        struct Charger *    charger     = App_BmsWorld_GetCharger(world);
-        struct Accumulator *accumulator = App_BmsWorld_GetAccumulator(world);
-        struct Airs *       airs        = App_BmsWorld_GetAirs(world);
+        struct BmsWorld *            world   = App_SharedStateMachine_GetWorld(state_machine);
+        struct Charger *             charger = App_BmsWorld_GetCharger(world);
+        struct Airs *                airs    = App_BmsWorld_GetAirs(world);
+        struct TractiveSystem *const ts      = App_BmsWorld_GetTractiveSystem(world);
 
-        static uint16_t ignore_chgr_fault_counter      = 0U;
-        const bool      is_charger_disconnected        = !App_Charger_IsConnected(charger);
-        bool            has_charger_faulted            = false;
-        bool            has_external_shutdown_occurred = !App_Airs_IsAirNegativeClosed(airs);
+        const bool external_shutdown_occurred = !App_Airs_IsAirNegativeClosed(airs);
+        const bool charging_enabled           = App_CanRx_Debug_ChargingSwitch_StartCharging_Get();
+        bool       charging_completed         = false;
+        const bool is_charger_connected       = App_Charger_IsConnected(charger);
 
-        if (ignore_chgr_fault_counter < CYCLES_TO_IGNORE_CHGR_FAULT)
+        const uint16_t ignore_chgr_fault_counter = App_Charger_GetFaultCounterVal(charger);
+        bool           has_charger_faulted       = false;
+        bool           timeout_passed            = false;
+
+        if (ignore_chgr_fault_counter >= CYCLES_TO_IGNORE_CHGR_FAULT)
         {
-            ignore_chgr_fault_counter++;
+            has_charger_faulted = App_Charger_HasFaulted(charger);
+            timeout_passed      = true;
+            UNUSED(ts);
+            // TODO: Re-implement charging_completed
+            //            if(-App_TractiveSystem_GetCurrent(ts) <=  0.85f && (App_TractiveSystem_GetCurrent(ts) < 0)){
+            //                charging_completed = true;
+            //            }
```

Comment:
charging_completed was magically going true immediately even though the current was much greater than 0.85, could be a current sense glitch, but CAN read the value as significantly higher than that (greater than 2.0A minimum during our tests if I recall correctly.) The current plan is just to babysit the charger and manually stop charging using the CAN message, but this should absolutely be looked into further in the future time permitting

Reply:
Not sure if you changed the code from what's in the commented out block, but if the ts current is positive when charging, the negative sign in front of the App_Tract... would be incorrect. But if you have the check for if ts current is negative as well, then the charging_completed = true should never be reached so I'm not sure.

Also, unrelated I realized we didn't use the CURRENT_AT_MAX_CHARGE variable here and just put 0.85f.

Reply:
Yeah the commented code was altered during testing, the reason we weren’t using the macro was a troubleshooting step, I’ll clean it up.

As for the negative sign in front of ts_current, it was reading as positive when it should have been negative, because the previous convention was to have power coming in be positive and it hasn’t fully been switched over yet. Again, I’ll clean up the commented block to be as you left it